flocker/acceptance/datahttp.py

```diff
@@ -0,0 +1,49 @@
+"""
+HTTP server that writes data to a specified file on POST, or reads and
+returns data from a specified file on GET.
+"""
+
+from urlparse import urlparse
+try:
+    from urlparse import parse_qs
+except ImportError:
+    from cgi import parse_qs
+
+from BaseHTTPServer import BaseHTTPRequestHandler, HTTPServer
+
+
+class Handler(BaseHTTPRequestHandler):
+    def do_POST(s):
+        length = int(s.headers['content-length'])
+        postvars = parse_qs(
+            s.rfile.read(length),
+            keep_blank_values=1
+        )
+        try:
+            with open(postvars["file"][0], "w") as f:
+                f.write(postvars["data"][0])
+        except Exception as e:
+            s.wfile.write(str(e.__class__) + ": " + str(e))
+        s.send_response(200)
+        s.end_headers()
+        s.wfile.write(b"ok")
+        s.wfile.close()
+
+    def do_GET(s):
+        s.send_response(200)
+        s.end_headers()
+        queryvars = parse_qs(urlparse(s.path).query)
```

Comment:
Could probably just always return the hardcoded file's contents. So test would fail if GET was called before POST since file would be missing.